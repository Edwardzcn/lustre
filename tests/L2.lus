/* Check ordering */


function f(x : int) returns (y : int)
  var c : int;
  var d : int;
let
  --%MAIN;
  c,d = g (x,x);
  y = c + d;
tel

function g(a : int; b : int) returns (c : int; d : int)
let
  c = a - b;
  d = a + b;
tel


