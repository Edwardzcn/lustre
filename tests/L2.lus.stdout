node (x_2 : int)
  returns (y_3)
  assumes ()
  shows ()
let
  a_10 = x_2
  b_11 = x_2
  c_4 = Sub(a_10, b_11)
  d_5 = Add(a_10, b_11)
  y_3 = Add(c_4, d_5)
tel
--- Step 1 ---
x_2 : int  = 1
y_3 = 2
--- Step 2 ---
x_2 : int  = 2
y_3 = 4
--- Step 3 ---
x_2 : int  = 3
y_3 = 6
--- Step 4 ---
x_2 : int  = 4
y_3 = 8
